src/query/plan/rule_based_planner.cpp

```diff
@@ -495,7 +495,7 @@ std::unique_ptr<LogicalOperator> GenReturnBody(std::unique_ptr<LogicalOperator>
   // Where may see new symbols so it comes after we generate Produce and in
   // general, comes after any OrderBy, Skip or Limit.
   if (body.where()) {
-    last_op = std::make_unique<Filter>(std::move(last_op), body.where()->expression_);
+    last_op = std::make_unique<Filter>(std::move(last_op), nullptr, body.where()->expression_);
```

Comment:
So `MATCH (n) RETURN n WHERE exists((n)-[]->())` is not a valid query, right? Is it enforced?

Reply:
I am not sure what should happen here, and what this part of the code does exactly, but it throws the same error as the following query:
`MATCH (n) RETURN n WHERE n.prop=1;`

And this is the error:
`mismatched input 'WHERE' expecting {<EOF>, ';'`

Reply:
@antaljanosbenjamin could you check this part. Why should this be valid query?

Reply:
Sorry, `WHERE` cannot go after `RETURN`, I was wrong here. I phrased my comment in a wrong way. I should have asked that we are sure we cannot use `WHERE exists(...)` in "return bodies", right? `RETURN` cannot have `WHERE` after it, but `WITH` can: `MATCH (n) WITH n WHERE n.prop=1 RETURN n`. I think I remember something to protect against that, but @Josipmrden could you please confirm?